pkg/ingest/ingest.go

```diff
@@ -0,0 +1,20 @@
+package ingest
+
+import (
+	"github.com/thanos-community/obslytics/pkg/dataframe"
+	"github.com/thanos-community/obslytics/pkg/input"
+)
+
+type Ingestor interface {
+	// Ingest a single series. It expects the series' chunks to be sorted
+	// by timestamp.
+	Ingest(input.Series) error
+
+	// Indicate no more Ingest calls to be done. It allows finalizing the active
+	// data even when not reaching next sample time.
+	Finalize() error
```

Comment:
We could put output.Writer her directly to avoid below `Process` glue (: 


Essentially I would try to simplify maybe a bit. Maybe it would be nice if Aggregator will literary take SeriesIterator and output.Writer directly? (: 

No need to ingester abstraction really :thinking:

Reply:
The reason I prefer having this outside is the ingestor/aggregator to focusing solely on converting the stream of series data into a data frame, not having to know anything about the how frequently to flush or how to iterate over the series.

This opens possibility to implement different transformations over those we did in aggregator: therefore the ingestor has quite generic name, while the aggregator is specific implementation that ingests the data to build aggregations.

To give you an example, different look at the data might be extracting continuous chunks for each series, where each row would represent specific time (start -> end) of some series, instead of doing the sampling/aggregation. Maybe it's premature generalization, but even without that, I still like a bit looser coupling here.

Therefore, I would suggest getting it as it's here, and experimenting with further changes outside of this PR to see what feels best.

Reply:
Agree, thanks!